benchmark/zk_census_benchmark_test.go

```diff
@@ -0,0 +1,179 @@
+	zkAddr, err := zk.NewRandAddress()
+	qt.Assert(b, err, qt.IsNil)
+
+	cparts := api.CensusParticipants{}
+	cparts.Participants = append(cparts.Participants, api.CensusParticipant{
+		Key:    zkAddr.Bytes(),
+		Weight: (*types.BigInt)(big.NewInt(1)),
+	})
+
+	_, code := cl.Request("POST", &cparts, censusID, "participants")
+	qt.Assert(b, code, qt.Equals, 200)
+
+	var resp []byte
+	censusData := &api.Census{}
+	resp, code = cl.Request("GET", nil, censusID, "root")
+	qt.Assert(b, code, qt.Equals, 200)
+	qt.Assert(b, json.Unmarshal(resp, censusData), qt.IsNil)
+	qt.Assert(b, censusData.Root, qt.IsNotNil)
+	root := censusData.Root
+
+	resp, code = cl.Request("GET", nil, censusID, "size")
+	qt.Assert(b, code, qt.Equals, 200)
+	qt.Assert(b, json.Unmarshal(resp, censusData), qt.IsNil)
+	b.Logf("root: %x | size: %d", root, censusData.Size)
+
+	// generate a proof
+	resp, code = cl.Request("GET", nil, censusID, "proof", zkAddr.String())
+	qt.Assert(b, code, qt.Equals, 200)
+	qt.Assert(b, json.Unmarshal(resp, censusData), qt.IsNil)
+	qt.Assert(b, censusData.Weight.String(), qt.Equals, "1")
+
+	censusData.Root = root
+	genProofZk(b, electionID, zkAddr, censusData)
+}
+
+func genProofZk(b *testing.B, electionID []byte, zkAddr *zk.ZkAddress, censusData *api.Census) {
+	// Get merkle proof associated to the voter key provided, that will contains
+	// the leaf siblings and value (weight)
+
+	log.Infow("zk census data received, starting to generate the proof inputs...", map[string]interface{}{
+		"censusRoot": censusData.Root, "electionId": fmt.Sprintf("%x", electionID)})
+
+	// Encode census root
+	strCensusRoot := arbo.BytesToBigInt(censusData.Root).String()
+	// Get vote weight
+	weight := new(big.Int).SetInt64(1)
+	if censusData.Weight != nil {
+		weight = censusData.Weight.MathBigInt()
+	}
+	// Get nullifier and encoded processId
+	nullifier, err := zkAddr.Nullifier(electionID)
+	qt.Assert(b, err, qt.IsNil)
+
+	// Calculate and encode vote hash and the encoded version of the electionID
+	strVoteHash := zk.BytesToArboStr(censusData.Value)
+	strProcessId := zk.BytesToArboStr(electionID)
+
+	// Unpack and encode siblings
+	unpackedSiblings, err := arbo.UnpackSiblings(arbo.HashFunctionPoseidon, censusData.Proof)
+	qt.Assert(b, err, qt.IsNil)
+
+	// Create a list of siblings with the same number of items that levels
+	// allowed by the circuit (from its config) plus one. Fill with zeros if its
+	// needed.
+	strSiblings := make([]string, zkCircuitTest.Levels+1)
+	for i := 0; i < len(strSiblings); i++ {
+		newSibling := "0"
+		if i < len(unpackedSiblings) {
+			newSibling = arbo.BytesToBigInt(unpackedSiblings[i]).String()
+		}
+		strSiblings[i] = newSibling
+	}
+	// Get artifacts of the current circuit
+	currentCircuit, err := circuit.LoadZkCircuit(context.Background(), zkCircuitTest)
+	qt.Assert(b, err, qt.IsNil)
+
+	// Create the inputs and encode them into a JSON
+	rawInputs := map[string]interface{}{
```

Comment:
uops, repeated struct definition.

Reply:
Sorry... Where? O.O

Reply:
apiclient/census.go L199

Reply:
Done! I created a new struct and function to simplify the circuit input generation: https://github.com/vocdoni/vocdoni-node/blob/feature/new_prover_integration/crypto/zk/circuit/inputs.go